### PR TITLE
fix: upd xml parser

### PR DESCRIPTION
**Pull request by olbychos**
> [!NOTE]
> **Medium Risk**
> Adjusts core agent XML parsing and auto-repairs tool-call JSON, which could change how some malformed model outputs are interpreted; covered by new integration tests but still affects a critical parsing path.
> 
> **Overview**
> Improves XML-mode tool-call robustness by making `XMLParser` tolerate common LLM formatting issues in `<action_input>`: it now repairs literal newlines inside JSON string values and unescapes HTML entities in parsed JSON so tools receive real characters like `<`, `>` and `&`.
> 
> Tightens XML preprocessing to better escape stray/unbalanced reserved tags (stack-based) and simplifies tag placeholder replacement to only target fully closed tags, reducing accidental partial matches. `Agent._handle_xml_mode` now converts `JSONParsingError` into a recoverable `ActionParsingException` with clearer guidance when `<action_input>` JSON is invalid, and prompts were updated to reflect the parser’s acceptance of plain vs XML-escaped symbols. Comprehensive integration tests were added for these edge cases (newlines, entities, nested structures, and recoverable agent errors).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit a6ddf5d19140a68e0e91687dcfe758c26344e82a. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by tyaroshko** [comment]
Do tests provide coverage for all cases with possible malformed input that we would like to handle?

**Comment by olbychos** [comment]
@tyaroshko / @maksymbuleshnyi please review it

File: dynamiq/nodes/agents/utils.py [hunk 1]
```diff
@@ -338,6 +349,55 @@ def _extract_data_lxml(
...
+                result.append(c)
+                escape_next = False
+                i += 1
+                continue
+            if c == "\\" and in_double_quote:
+                escape_next = True
+                result.append(c)
+                i += 1
+                continue
+            if c == '"' and not escape_next:
```
Comment (tyaroshko) [inline dynamiq/nodes/agents/utils.py:375]: The check for `escape_next` is redundant

Comment (acoola) [reply dynamiq/nodes/agents/utils.py:375]: let's create fix later
